plugins/wacom-usb/fu-wac-module-bluetooth-id9.c

```diff
@@ -0,0 +1,267 @@
+/*
+ * Copyright (C) 2018 Richard Hughes <richard@hughsie.com>
+ * Copyright (C) 2021-2023 Jason Gerecke <jason.gerecke@wacom.com>
+ *
+ * SPDX-License-Identifier: LGPL-2.1+
+ */
+
+#include "config.h"
+
+#include <string.h>
+
+#include "fu-wac-common.h"
+#include "fu-wac-device.h"
+#include "fu-wac-module-bluetooth-id9-struct.h"
+#include "fu-wac-module-bluetooth-id9.h"
+#include "fu-wac-struct.h"
+
+struct _FuWacModuleBluetoothId9 {
+	FuWacModule parent_instance;
+};
+
+G_DEFINE_TYPE(FuWacModuleBluetoothId9, fu_wac_module_bluetooth_id9, FU_TYPE_WAC_MODULE)
+
+#define FU_WAC_MODULE_BLUETOOTH_ID9_PAYLOAD_SZ	   256
+#define FU_WAC_MODULE_BLUETOOTH_ID9_START_NORMAL   0x00
+#define FU_WAC_MODULE_BLUETOOTH_ID9_CMD_NORMAL	   0x00
+#define FU_WAC_MODULE_BLUETOOTH_ID9_CMD_FULLERASE  0xFE
+#define FU_WAC_MODULE_BLUETOOTH_ID9_LOADER_RAM	   0x02
+#define FU_WAC_MODULE_BLUETOOTH_ID9_LOADER_BEGIN   0x03
+#define FU_WAC_MODULE_BLUETOOTH_ID9_LOADER_DATA	   0x04
+#define FU_WAC_MODULE_BLUETOOTH_ID9_CRC_POLYNOMIAL 0xEDB88320
+
+static GBytes *
+fu_wac_module_bluetooth_id9_get_startcmd(const guint8 *data,
+					 gsize sz,
+					 gboolean full_erase,
+					 GError **error)
+{
+	g_autoptr(GByteArray) loader_cmd = fu_struct_loader_cmd_new();
+	g_autoptr(GByteArray) spi_cmd = fu_struct_spi_cmd_new();
+	g_autoptr(GByteArray) unknown_cmd = fu_struct_unknown_cmd_new();
```

Comment:
normally the g_auto lines go last in the variable block

Reply:
Would these be at the end of the block (i.e. after the intentionally-uninitialized crc variable) or at the end of the block of initialized variables?

Reply:
after guint32 crc please.

Reply:
Ack